core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala

```diff
@@ -99,12 +100,22 @@ class SnappyStoreHiveCatalog(externalCatalog: SnappyExternalCatalog,
 //  }
 
   protected var currentSchema: String = {
-    val defaultName = Constant.DEFAULT_SCHEMA
+    var user = snappySession.conf.get(Attribute.USERNAME_ATTR, "")
+    val defaultName = if (user.isEmpty) {
+      // In smart connector, property name is different.
+      user = snappySession.conf.get(Constant.SPARK_STORE_PREFIX + Attribute.USERNAME_ATTR, "")
+      if (user.isEmpty) Constant.DEFAULT_SCHEMA else formatDatabaseName(user)
+    } else {
+      formatDatabaseName(user)
+    }
     val defaultDbDefinition =
       CatalogDatabase(defaultName, "app database", sqlConf.warehousePath, Map())
-    // Initialize default database if it doesn't already exist
-    externalCatalog.createDatabase(defaultDbDefinition, ignoreIfExists = true)
-    client.setCurrentDatabase(defaultName)
+    SnappyContext.getClusterMode(snappySession.sparkContext) match {
+      case ThinClientConnectorMode(_, _) =>
+      case _ =>     // Initialize default database if it doesn't already exist
+        externalCatalog.createDatabase(defaultDbDefinition, ignoreIfExists = true)
+        client.setCurrentDatabase(defaultName)
+    }
```

Comment:
Probably the initialization code of currentSchema can be arranged differently
In case of ThinClientConnectorMode(_, _),  get user as required by using Constant.SPARK_STORE_PREFIX property in the other case get it using just Attribute.USERNAME_ATTR property. 
Also in the current method formatDatabaseName() gets called twice if the 'user' is set (that is not Constant.DEFAULT_SCHEMA)